### PR TITLE
Fix comparing of sha512sums

### PR DESCRIPTION
**Pull request by jelly**
read does not read from a pipe, only if lastpipe is set.

### CODE REVIEW

File: repro.in [hunk 1]
```diff
@@ -374,8 +374,8 @@ __END__
     build_package "build" $builddir
 
     msg "Comparing hashes..."
-    sha512sum -b "$pkg" | read refference_build_checksum _
-    sha512sum -b ./build/"$(basename $pkgfile)" | read other_build_checksum _
+    read -r refference_build_checksum _ < <(sha512sum -b "$pkg")
+    read -r other_build_checksum _ < <(sha512sum -b ./build/"$(basename $pkgfile)")
     if [ "$refference_build_checksum" = "$other_build_checksum" ]; then
```
Comment (kpcyrd) [inline repro.in:379]: I think the proper bash way is probably:
```sh
msg "Comparing files..."
if diff -q -- "$pkg" ./build/"$(basename $pkgfile)" > /dev/null ; then
  msg "Package is reproducible!"
else
[...]
```

Comment (Foxboron) [reply repro.in:379]: Hum. Gotta try out that and see what i like better. Thanks!

Comment (kpcyrd) [reply repro.in:379]: Also:
```sh
./build/"$(basename $pkgfile)"
```
is buggy and should be
```
./build/"$(basename "$pkgfile")"
```
as it currently doesn't handle spaces in $pkgfile correctly.

Comment (jelly) [reply repro.in:379]: I have no preference tbh, @Foxboron you might want to move this code to a separate function so it's re-usable for build and check.

**Comment by Foxboron** [comment]
I implemented the diff approach as suggested by @kpcyrd with https://github.com/archlinux/archlinux-repro/commit/5ff27cf8d18db40e5575d623e5522216edb6132a

**Pull request closed without merging**